modmail/extensions/extension_manager.py

```diff
@@ -0,0 +1,278 @@
+
+
+BASE_PATH_LEN = __name__.count(".")
+
+EXT_METADATA = ExtMetadata(load_if_mode=BotModes.DEVELOP, no_unload=True)
+
+
+class Action(Enum):
+    """Represents an action to perform on an extension."""
+
+    # Need to be partial otherwise they are considered to be function definitions.
+    LOAD = functools.partial(ModmailBot.load_extension)
+    UNLOAD = functools.partial(ModmailBot.unload_extension)
+    RELOAD = functools.partial(ModmailBot.reload_extension)
+
+
+class ExtensionConverter(commands.Converter):
+    """
+    Fully qualify the name of an extension and ensure it exists.
+
+    The * and ** values bypass this when used with the reload command.
+    """
+
+    source_list = EXTENSIONS
+    type = "extension"
+
+    async def convert(self, ctx: Context, argument: str) -> str:
+        """Fully qualify the name of an extension and ensure it exists."""
+        # Special values to reload all extensions
+        if argument == "*" or argument == "**":
+            return argument
+
+        argument = argument.lower()
+
+        if argument in self.source_list:
+            return argument
+
+        qualified_arg = f"modmail.{self.type}s.{argument}"
+        if qualified_arg in self.source_list:
+            return qualified_arg
+
+        matches = []
+        for ext in self.source_list:
+            if argument == unqualify(ext):
+                matches.append(ext)
+
+        if not matches:
+            raise commands.BadArgument(f":x: Could not find the {self.type} `{argument}`.")
+
+        if len(matches) > 1:
+            names = "\n".join(sorted(matches))
+            raise commands.BadArgument(
+                f":x: `{argument}` is an ambiguous {self.type} name. "
+                f"Please use one of the following fully-qualified names.```\n{names}```"
+            )
+
+        return matches[0]
+
+
+class ExtensionManager(ModmailCog, name="Extension Manager"):
+    """Extension management base class."""
+
+    type = "extension"
+
+    def __init__(self, bot: ModmailBot):
+        self.bot = bot
+        self.all_extensions = EXTENSIONS
+
+    def get_black_listed_extensions(self) -> list:
+        """Returns a list of all unload blacklisted extensions."""
+        return NO_UNLOAD
+
+    @commands.group("ext", aliases=("extensions", "exts"), invoke_without_command=True)
+    async def extensions_group(self, ctx: Context) -> None:
+        """Load, unload, reload, and list loaded extensions."""
+        await ctx.send_help(ctx.command)
+
+    @extensions_group.command(name="load", aliases=("l",))
+    async def load_extensions(self, ctx: Context, *extensions: ExtensionConverter) -> None:
+        """
+        Load extensions given their fully qualified or unqualified names.
+
+        If '\*' or '\*\*' is given as the name, all unloaded extensions will be loaded.
+        """  # noqa: W605
+        if not extensions:
+            await ctx.send_help(ctx.command)
+            return
```

Comment:
This is against our guru the holy PEP8, if this isn't solved bad deeds will happen with modmail 🆘 

```suggestion
            return None
```

> Be consistent in return statements. Either all return statements in a function should return an expression, or none of them should. If any return statement returns an expression, any return statements where no value is returned should explicitly state this as `return None`, and an explicit return statement should be present at the end of the function (if reachable)

Reply:
> Either all return statements in a function should return an expression, or none of them should.

None of the other return statements in this function return a value either.

Reply:
I had take it as a complete decision though to maintain consistency throughout the repository. Either we keep it or not, doesn't really matter for me.